### ISSUE TITLE
`arguments` and `eval` should be allowed in the return position in strict functions

### ISSUE THREAD

**Issue by raskad** [boa-dev/boa#1598]
In strict mode `arguments` and `eval` are mostly treated as keywords. For example they cannot be assigned to. But they should be allowed as return values from a strict function.

The following code examples currently fail, but should work:

```javascript
let eval = 1;

function a(){
    "use strict";
    return eval;
}

a() // returns 1
```

```javascript
function a(){
    "use strict";
    return arguments;
}

a(1,2) // returns [Arguments] { '0': 1, '1': 2 }
```

Based on the error message a good starting point would be here: https://github.com/boa-dev/boa/blob/c977b8e5aa9634ff0e804daef7510feca69a3d10/boa/src/syntax/lexer/identifier.rs#L119

**Comment by am-a-man** [comment]
hey,
I would like to work on this, could you assign me this issue?

**Comment by Razican** [comment]
@am-a-man Do you have any updates on the progress of this?

**Comment by am-a-man** [comment]
@Razican sorry for the long delay, will try to complete it by the end of this week

**Comment by raskad** [comment]
Fixed by #2069
